File: app-fw/app-base/src/main/java/org/yy/base/dao/dynamic2/AndExpression.java
package org.yy.base.dao.dynamic2;

import java.util.ArrayList;
import java.util.List;


    public class AndExpression extends Expression
    {
        public  String Eval(ExpressionContext ctx)
        {
            if (lists.size() == 0)
            {
                return "";
            }

            StringBuilder sb = new StringBuilder();
            List<String> results = new ArrayList<String>();
            for (int i = 0; i < lists.size(); i++)
            {
                Expression ex = lists.get(i);
                String result = ex.Eval(ctx);
                if (result != null && result.trim().length() > 0)
                {
                    results.add(result);
                }
            }

            if (results.size() == 0)
            {
                return "";
            }
            for (int i = 0; i < results.size(); i++)
            {
                sb.append("(");
                sb.append(results.get(i));
                sb.append(") ");
                if (i != results.size() - 1)
                {
                    sb.append(" AND ");
                }
            }

            return sb.toString();
        }
    }

File: SqlParser/SqlParser/AndExpression.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SqlParser
{
    public class AndExpression : Expression
    {
        public override string Eval(IExpressionContext ctx)
        {
            if (lists.Count == 0)
            {
                return "";
            }

            StringBuilder sb = new StringBuilder();
            List<String> results = new List<string>();
            for (int i = 0; i < lists.Count; i++)
            {
                Expression ex = lists[i];
                string result = ex.Eval(ctx);
                if (result != null && result.Trim().Length > 0)
                {
                    results.Add(result);
                }
            }

            if (results.Count == 0)
            {
                return "";
            }
            for (int i = 0; i < results.Count; i++)
            {
                sb.Append("(");
                sb.Append(results[i]);
                sb.Append(") ");
                if (i != results.Count - 1)
                {
                    sb.Append(" AND ");
                }
            }

            return sb.ToString();
        }
    }
}
